### PR TITLE
Document build.info syntax internally

### PR DESCRIPTION
**Pull request by levitte**

### CODE REVIEW

**Comment by levitte** [comment]
This has been due for a very long time.

File: doc/internal/man7/build.info.pod [hunk 1]
```diff
@@ -0,0 +1,491 @@
...
+=item How to build it?
+
+This is about build file generation.
+
+=back
+
+This document is all about the second item, "What to build?", and most
+of all, how to specify that information.
+
+I<for some terms used in this document, please see the glossary at the
```
Comment (richsalz) [inline doc/internal/man7/build.info.pod]: I don't think the ```I<>``` markup is needed.  Start with uppercase For.  Use an ```L</glossary>``` link.

Comment (levitte) [reply doc/internal/man7/build.info.pod]: Sure.

Comment (levitte) [reply doc/internal/man7/build.info.pod]: Done

**Comment by levitte** [comment]
```
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
doc/internal/man7/build.info.pod:1: period in =head
```

Argh!  This is man7, which is largely prose.  Meh :-(

Although, it might be a bit much to use =head for enumeration, so I'll change that and satisfy the checks as a bonus.

**Comment by richsalz** [comment]
The "exdata" manpage put the items as a code block FWIW.

**Comment by levitte** [comment]
> The "exdata" manpage put the items as a code block FWIW.

That's not what I meant, but have a look now and you'll see the difference.

Side note: the exdata page should not be taken as an example of a well written manpage.  I've some thoughts of rewriting it at some point, if no one else gets there before me.

**Comment by richsalz** [comment]
> the exdata page should not be taken as an example of a well written manpage.

Hey! :)  It was the first generic manpage, merged from all the other per-type manpages.

**Comment by levitte** [comment]
> Hey! :) It was the first generic manpage, merged from all the other per-type manpages.

Your sources worked against you, I'm afraid ;-)

File: doc/internal/man7/build.info.pod [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+=item B<ELSIF[>0|1B<]>
+
+=item B<ELSE>
+
+=item B<ENDIF>
+
+=back
+
+This works like any condition system with similar syntax, and the
+condition value in B<IF> and B<ELSIF> can really literal value that
```
Comment (t8m) [inline doc/internal/man7/build.info.pod]: I cannot parse this - seems some word(s) is missing?

File: doc/internal/man7/build.info.pod [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+=item B<$>I<NAME> B<=> I<value>
+
+=back
+
+These variables can then be used as part of any statement value or
+indexed statement item.  This should be used with some care, as
+I<variables are expanded into their values before the value they are
+part of is tokenized>.
+
+I<variable assignment values are not tokenized.>
```
Comment (t8m) [inline doc/internal/man7/build.info.pod]: Uppercase V.

File: doc/internal/man7/build.info.pod [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+F<build.info> variables and B<SUBDIRS> statement, for which the scope
+is the F<build.info> file they are in.
+
+=head2 Perl nuggets
+
+Whenever a F<build.info> file is read, it is passed through the Perl
+template processor L<OpenSSL::Template>, which is a small extension of
+L<Text::Template>.
+
+Perl nuggest are anything between C<{-> and C<-}>, and whatever the
```
Comment (t8m) [inline doc/internal/man7/build.info.pod]: nuggets

File: doc/internal/man7/build.info.pod [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+over.
+
+If the condition is false (represented as C<0> here), everything
+from this B<IF> is skipped over until the next corresponding B<ELSIF>
+or B<ELSE>, at which point processing continues.
+
+=item B<ELSE>
+
+If F<build.info> statements have been skipped over to this point since
+the corresponding B<IF> or B<ELSIF>, F<build.info> processing start
```
Comment (t8m) [inline doc/internal/man7/build.info.pod]: starts

**Review by t8m — CHANGES_REQUESTED** [review]
A few nits.

File: doc/internal/man7/build.info.pod (hunk 2 above)
Comment (levitte) [reply doc/internal/man7/build.info.pod]: Yup.  Looking at it now, I was myself wonder what the hell I meant, but a second read made it clear.

Fixed

File: doc/internal/man7/build.info.pod (hunk 3 above)
Comment (levitte) [reply doc/internal/man7/build.info.pod]: Yup

File: doc/internal/man7/build.info.pod (hunk 4 above)
Comment (levitte) [reply doc/internal/man7/build.info.pod]: Fixed

File: doc/internal/man7/build.info.pod (hunk 5 above)
Comment (levitte) [reply doc/internal/man7/build.info.pod]: Corrected

**Review by t8m — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

e20ba0a51571ce48043e129546deb15be30f287d Document build.info syntax internally

**Pull request closed without merging**